app/jobs/dhis2/ethiopia_exporter_job.rb

```diff
@@ -1,23 +1,111 @@
 module Dhis2
   class EthiopiaExporterJob < Dhis2ExporterJob
+    REQUIRED_GENDERS = ["male", "female"]
+    AGE_RANGES = [[18, 29], [30, 39], [40, 69], [70, 200]]
+    MONTHS_FOR_COHORT_QUERY = 6
+
     private
 
     def facility_data_for_period(region, period)
+      cohort_registered_patients = PatientStates::Hypertension::RegistrationsForMonthsQuery.new(region, period, MONTHS_FOR_COHORT_QUERY).call
       {
-        htn_cumulative_assigned: PatientStates::Hypertension::CumulativeAssignedPatientsQuery.new(region, period).call.count,
-        htn_controlled: PatientStates::Hypertension::ControlledPatientsQuery.new(region, period).call.count,
-        htn_uncontrolled: PatientStates::Hypertension::UncontrolledPatientsQuery.new(region, period).call.count,
-        htn_missed_visits: PatientStates::Hypertension::MissedVisitsPatientsQuery.new(region, period).call.count,
-        htn_ltfu: PatientStates::Hypertension::LostToFollowUpPatientsQuery.new(region, period).call.count,
-        htn_dead: PatientStates::Hypertension::DeadPatientsQuery.new(region, period).call.count,
-        htn_cumulative_registrations: PatientStates::Hypertension::CumulativeRegistrationsQuery.new(region, period).call.count,
-        htn_monthly_registrations: PatientStates::Hypertension::MonthlyRegistrationsQuery.new(region, period).call.count,
-        htn_cumulative_assigned_adjusted: PatientStates::Hypertension::AdjustedAssignedPatientsQuery.new(region, period).call.count
+        htn_enrolled_under_care: {category_option_key: "dhis2_age_gender_category_elements", values: format_gender_age_data(PatientStates::Hypertension::RegistrationsUnderCareGenderAgeAggregatedQuery.new(region, period).call)},
+        htn_enrolled_under_care_treatment: {category_option_key: "dhis2_treatment_category_elements", values: format_treatment_data(PatientStates::Hypertension::RegistrationsUnderCareQuery.new(region, period).call)},
+        htn_by_enrollment_time: {category_option_key: "dhis2_enrollment_time_category_elements", values: format_enrollment_data(PatientStates::Hypertension::CumulativeRegistrationsQuery.new(region, period).call)},
+        htn_cohort_registered: {category_option_key: "dhis2_cohort_registered_category_elements", values: format_cohort_registered_data(cohort_registered_patients)},
+        htn_cohort_outcome: {category_option_key: "dhis2_cohort_category_elements", values: format_cohort_data(cohort_registered_patients)}
       }
     end
 
     def data_elements_map
       CountryConfig.dhis2_data_elements.fetch(:dhis2_data_elements)
     end
+
+    def get_category_element_mappings(category_element_key)
+      CountryConfig.dhis2_data_elements.fetch(category_element_key.to_sym)
+    end
+
+    def get_attiribute_option_mapping
```

Comment:
[nitpick, nonblocking] This seems to return a singular ID and not a mapping. How about renaming this to `attribute_option_id`.

So the call site would look like this:
```ruby
              attribute_option_combo: attribute_option_id,
```

no need to "get" it into a variable that is used just once that way.

Reply:
similarly the method above this could be named `category_element_map`